resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/MountVolumesFeatureStep.scala

```diff
@@ -85,6 +85,7 @@ private[spark] class MountVolumesFeatureStep(conf: KubernetesConf)
               .withApiVersion("v1")
               .withNewMetadata()
                 .withName(claimName)
+                .addToLabels(SPARK_APP_ID_LABEL, conf.sparkConf.getAppId)
```

Comment:
I am trying to understand this ... will the sparkConf.getAppId be available at this point ?

Reply:
Yes, `Spark Driver` pod is already launched in the K8s and  the driver is building executor pod specs here.